prog/storage/setup_spdk.rb

```diff
@@ -34,19 +34,30 @@ def self.assemble(vm_host_id, version, start_service: false, allocation_weight:
     SpdkInstallation.create(
       version: frame["version"],
       allocation_weight: 0,
-      vm_host_id: vm_host.id
+      vm_host_id: vm_host.id,
+      cpu_count: spdk_cpu_count(total_host_cpus: vm_host.total_cpus)
     ) { _1.id = SpdkInstallation.generate_uuid }
 
     hop_install_spdk
   end
 
   label def install_spdk
-    version = frame["version"]
-    sshable.cmd("sudo host/bin/setup-spdk install #{version.shellescape}")
+    q_version = frame["version"].shellescape
+    cpu_count = spdk_cpu_count(total_host_cpus: vm_host.total_cpus)
+    sshable.cmd("sudo host/bin/setup-spdk install #{q_version} #{cpu_count}")
 
     hop_start_service
   end
 
+  def spdk_cpu_count(total_host_cpus:)
+    # Use 4 SPDK cpus on AX162, 2 elsewhere.
+    if total_host_cpus == 96
+      4
+    else
+      2
+    end
```

Comment:
It's a bit oblique and not adaptive to other sizes, is it better if we banded things using a formula, to keep things roughly in proportion? While there's little doubt we'll have to adjust this as we figure out the span of computers we practically want to run on, it's pretty weird that if I run this on a 128 core leaseweb computer I drop back to one core from two.

Reply:
I changed the formula to allocate 2 spdk cpus per each 64 host cpus, so we'll have 2 for 1..64, 4 for 65..128.